components/admin_console/team_channel_settings/group/group_users/users_to_remove.tsx

```diff
@@ -239,89 +243,85 @@ export default class UsersToRemove extends React.PureComponent<Props, State> {
-            filterOptions.role.values = {
-                ...filterOptions.role.values,
-                [GeneralConstants.CHANNEL_USER_ROLE]: {
-                    name: (
-                        <FormattedMessage
-                            id='admin.user_item.member'
-                            defaultMessage='Member'
-                        />
-                    ),
-                    value: false,
-                },
-                [GeneralConstants.CHANNEL_ADMIN_ROLE]: {
-                    name: (
-                        <FormattedMessage
-                            id='admin.user_grid.channel_admin'
-                            defaultMessage='Channel Admin'
-                        />
-                    ),
-                    value: false,
-                },
-            };
-            filterOptions.role.keys = [GeneralConstants.SYSTEM_GUEST_ROLE, GeneralConstants.CHANNEL_USER_ROLE, GeneralConstants.CHANNEL_ADMIN_ROLE, GeneralConstants.SYSTEM_ADMIN_ROLE];
+            roleFilter.values.set(GeneralConstants.CHANNEL_USER_ROLE, {
+                name: (
+                    <FormattedMessage
+                        id='admin.user_item.member'
+                        defaultMessage='Member'
+                    />
+                ),
+                value: false,
+            });
+            roleFilter.values.set(GeneralConstants.CHANNEL_ADMIN_ROLE, {
+                name: (
+                    <FormattedMessage
+                        id='admin.user_grid.channel_admin'
+                        defaultMessage='Channel Admin'
+                    />
+                ),
+                value: false,
+            });
+
+            roleFilter.keys = [GeneralConstants.SYSTEM_GUEST_ROLE, GeneralConstants.CHANNEL_USER_ROLE, GeneralConstants.CHANNEL_ADMIN_ROLE, GeneralConstants.SYSTEM_ADMIN_ROLE];
         } else if (this.props.scope === 'team') {
-            filterOptions.role.values = {
-                ...filterOptions.role.values,
-                [GeneralConstants.TEAM_USER_ROLE]: {
-                    name: (
-                        <FormattedMessage
-                            id='admin.user_item.member'
-                            defaultMessage='Member'
-                        />
-                    ),
-                    value: false,
-                },
-                [GeneralConstants.TEAM_ADMIN_ROLE]: {
-                    name: (
-                        <FormattedMessage
-                            id='admin.user_grid.team_admin'
-                            defaultMessage='Team Admin'
-                        />
-                    ),
-                    value: false,
-                },
-            };
-            filterOptions.role.keys = [GeneralConstants.SYSTEM_GUEST_ROLE, GeneralConstants.TEAM_USER_ROLE, GeneralConstants.TEAM_ADMIN_ROLE, GeneralConstants.SYSTEM_ADMIN_ROLE];
+            roleFilter.values.set(GeneralConstants.TEAM_USER_ROLE, {
```

Comment:
Adding and removing elements now is also a bit different. I feel it is clearer, but not sure.